lib/ngx/re.lua

```diff
@@ -168,6 +152,24 @@ function _M.split(subj, regex, opts, ctx, max, res)
                 break
             end
 
+            if from == to then
+                local old_pos = ctx.pos
+
+                local from2, _, _, err2 = re_split_helper(subj, compiled,
+                                              compile_once, flags, ctx)
+                if err then
+                    return nil, err2
+                end
+
+                if not from2 then
+                    break
+                end
+
+                from = from2
+                to = from2
+                ctx.pos = old_pos
```

Comment:
Hmmm yeah I thought this was already the case (edited my explanatory comment), but just realized it actually is backtracking to the old position. Will try to get rid of it. Would want nothing less performant than that for lua-resty-core :)